libsolidity/codegen/Compiler.cpp

```diff
@@ -49,6 +49,9 @@ void Compiler::compileContract(
 	m_runtimeSub = creationCompiler.compileConstructor(_contract, _otherCompilers);
 
 	m_context.optimise(m_optimiserSettings);
+
+	solAssert(m_context.requestedYulFunctionsRan(), "requestedYulFunctions() was not called.");
```

Comment:
Does that work here? Shouldn't we rather have it in `Compiler::compileContract`?

Reply:
As it is currently, that would cause a assert fail because we do `m_runtimeCompiler->appendMissingFunctions()` when we call `compileConstructor()` which adds it to the runtime context.

But this is a bit confusing because only the call to the creation compiler adds the missing functions to the runtime compiler.. maybe we should structure this differently?